### ISSUE TITLE
Use Entry match accuracy setting over Domain overrides if more strict

### ISSUE THREAD

**Issue by filviu** [kee-org/keepassrpc#97]
I hope I can explain this properly. But first thank you for Kee, it's great !

Use case: 

* My database default is "Domain", this works best for general sites, online stores, etc.
* For some work related domains I've setup a domain override as "hostname" (because jira.example.com and mail.example.com and so on)

But on that work domain I also have items where I want exact matching say pfsense.example.com or jenkins.example.com as it's very easy to break a config in those tools as they have pages long forms and it's easy to miss a spot where it auto-filled.

It took a while to realise that the Domain override trumps entry setting. Which in a way makes sense as you can't know if the entry setting is a historical default or something the user set.

What I'm proposing is to have the entry win if it's more strict than the domain setting. 

TL;DR if domain override is below the entry setting use the entry setting on the scale Domain->Hostname->Exact

**Comment by luckyrat** [comment]
I don't think this issue is specific/unique to KeePassRPC. If you're still interested in improvements related to this issue, please use the [Kee community forum](https://forum.kee.pm/) and provide the above information so others can help discuss, evaluate and move this idea forward. In future, using the community forum will likely result in faster help (I've just added a GitHub issue template to make that clearer for people arriving at this repo from routes that don't already mention that).
